activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -119,7 +134,7 @@ class Array
     alias :orig_sum :sum
 
     def sum(init = nil, &block) #:nodoc:
-      if init.is_a?(Numeric) || first.is_a?(Numeric)
+      if init || first.is_a?(Numeric)
```

Comment:
We could also take the opportunity to remove AS `#sum` entirely and rely on Ruby 2.4+ behavior for the future. It's so close to AS sum now; the only difference is requiring the initial value.

We could deprecate calling `#sum` on a non-Numeric Enumerable without an initial arg, then break compat in 5.1, and remove our `#sum` entirely in Rails 6.